### PR TITLE
Fix flaky basic auth dialer test

### LINKED ISSUE #14143: `TestALPNProxyHTTPProxyBasicAuthDial` flakiness
#### Link(s) to logs

- https://console.cloud.google.com/cloud-build/builds;region=us-west1/391f6966-8d99-4d9d-a1d5-a2c1753555bc;step=0?project=ci-account
- https://console.cloud.google.com/cloud-build/builds;region=us-west1/365fbf4d-9061-4204-9d82-95c88bffb373?project=ci-account
- https://console.cloud.google.com/cloud-build/builds/c96fa16b-3f09-433a-aaef-07b08a585650?project=ci-account

#### Relevant snippet

```
    proxy_test.go:886: 
        	Error Trace:	proxy_test.go:886
        	Error:      	Received unexpected error:
        	            	did not find 1 nodes
        	Test:       	TestALPNProxyHTTPProxyBasicAuthDial
```

### PR DESCRIPTION
**Pull request by GavinFrazar**
Closes #14143

In my last attempt to fix this test, I introduced a subtle race condition in the test code. Since the ProxyAuthorizer was not blocking when it sends the request error over the channel, it was racing the "waiter". If the waiter had not started receiving yet, then the error would never be sent and the waiter would time out, failing the test.

I fixed that by using a buffered channel.

While I was looking into the cause of the new failure I also realized we could speed up this test a lot:
1. The test would start a node and let it fail to register initially, but in doing so the reconnect backoff time would go higher and higher (getting up to 30-40 seconds).
2. Then we gave the node correct basic auth credentials and waited for it to successfully register, but it would take some time to actually register.

To speed up the test, I instead start the node in a separate goroutine and wait for a request to come into the ProxyAuthorizer, reject it for bad auth, then make the auth credentials correct and let the next reconnect attempt be authorized. The test runs in about 5 seconds now.

### CODE REVIEW

File: integration/proxy/proxy_test.go [hunk 1]
```diff
@@ -1116,7 +1116,6 @@ func TestALPNProxyHTTPProxyBasicAuthDial(t *testing.T) {
 	}
 	cfg.Listeners = helpers.SingleProxyPortSetupOn(rcAddr)(t, &cfg.Fds)
 	rc := helpers.NewInstance(t, cfg)
-	defer rc.StopAll()
```
Comment (zmb3) [inline integration/proxy/proxy_test.go:1119]: Why remove this?

Comment (GavinFrazar) [reply integration/proxy/proxy_test.go:1119]: Realized it was already deferred below after `rc.Start()`, but I guess I should remove the one below not this one

File: integration/proxy/proxy_test.go [hunk 2]
```diff
@@ -1162,19 +1161,25 @@ func TestALPNProxyHTTPProxyBasicAuthDial(t *testing.T) {
 	t.Setenv("http_proxy", helpers.MakeProxyAddr(user, pass, proxyURL.Host))
 
 	rcProxyAddr := net.JoinHostPort(rcAddr, helpers.PortStr(t, rc.Web))
-	require.Zero(t, ph.Count())
 	nodeCfg := makeNodeConfig("node1", rcProxyAddr)
 	nodeCfg.Log = log
-	_, err = rc.StartNode(nodeCfg)
-	require.Error(t, err)
 
+	require.Zero(t, ph.Count())
```
Comment (ibeckermayer) [inline integration/proxy/proxy_test.go:1167]: Seems not worth checking for until the node has been started?

Comment (GavinFrazar) [reply integration/proxy/proxy_test.go:1167]: Sanity check, but yes it should always be 0 I can remove

File: integration/proxy/proxy_test.go (hunk 1 above)
Comment (GavinFrazar) [reply integration/proxy/proxy_test.go]: Restored this and removed the one below

**Review by ibeckermayer — APPROVED** [review]

**Review by jakule — APPROVED** [review]

**Pull request merged**